src/Generator.hack

```diff
@@ -201,7 +201,7 @@ class HTMLPurifier_Generator {
     public function escape(string $string, int $quote = 0): string {
         // Workaround for APC bug on Mac Leopard reported by sidepodcast
         // http://htmlpurifier.org/phorum/read.php?3,4823,4846
-        if ($quote is null) {
+        if ($quote == 0) {
```

Comment:
Should be `===`?  What happens if `$quote` _is_ null?

Reply:
`$quote` has a default value of 0 in the function argument. I have updated it to -1 https://www.php.net/manual/en/function.htmlspecialchars.php